hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -1592,17 +1592,22 @@ private PersistedUserVolumeInfo getVolumesByUser(String userNameKey)
    * @throws IOException
    */
   public PendingKeysDeletion getPendingDeletionKeys(final int keyCount,
-                             OmSnapshotManager omSnapshotManager)
+      final String startKey, OmSnapshotManager omSnapshotManager)
       throws IOException {
     List<BlockGroup> keyBlocksList = Lists.newArrayList();
     HashMap<String, RepeatedOmKeyInfo> keysToModify = new HashMap<>();
+    String lastKey = "";
     try (TableIterator<String, ? extends KeyValue<String, RepeatedOmKeyInfo>>
              keyIter = getDeletedTable().iterator()) {
+      if (!Strings.isNullOrEmpty(startKey)) {
+        keyIter.seek(startKey);
```

Comment:
We should eliminate this seek() and specify startKey in the iterator constructor to avoid a seekToFirst() call.